app/services/auth_token_service.rb

```diff
@@ -0,0 +1,22 @@
+class AuthTokenService
+
+  def self.generate(auth_hash, request_host)
+    access_key = nil
+    recipient_email = auth_hash['recipient_email']
+
+    auth_token = AuthToken.new(auth_hash).generate
+
+    if recipient_email
+      # generate a seperate authentication for the recipient
+      AuthToken.new(email:  recipient_email).generate
+
+      # cretae an empty secret
+      access_key = SecureRandom.hex(10)
+      SecretService.create_empty_secret(auth_hash['email'],
```

Comment:
Creating a Secret in the authToken service/controller seems like a tight coupling and a mix of responsibility.
At minimum, there is a naming problem: `AuthTokenService.generate` doesn't suggest in any way that I will also end up with a secret.
The check for recipient_email and the creation of the secret should maybe move to the controller, making it more obvious.

Better, I think the creation of the Secret should generate the Token for the recipient, if needed (which will always be the case, but just happening at different moment depending on the path)